pkg/syncer/broker/syncer.go

```diff
@@ -275,11 +275,15 @@ func createBrokerClient(config *SyncerConfig) error {
 
 		config.BrokerNamespace = spec.RemoteNamespace
 
+		// If we have a secret, try to use it
 		if spec.Secret != "" {
 			config.BrokerRestConfig, authorized, err = resource.GetAuthorizedRestConfigFromFiles(spec.APIServer,
 				filepath.Join(SecretPath(spec.Secret), "token"), filepath.Join(SecretPath(spec.Secret), "ca.crt"),
 				&rest.TLSClientConfig{Insecure: spec.Insecure}, *gvr, spec.RemoteNamespace)
-		} else {
+		}
+
+		// If we encountered an error, or we don't have a secret, use the values in the spec
+		if spec.Secret == "" || err != nil {
```

Comment:
Should we print a warning in this case instead of discarding the error out right?

Otherwise we might miss some other situations not related to the original bug

Reply:
Unfortunately we don’t have access to a logger in this part of the code, AFAICT.

Reply:
I see `klog` imported and used?